datadog_checks_dev/datadog_checks/dev/tooling/commands/env/test.py

```diff
@@ -0,0 +1,69 @@
+# (C) Datadog, Inc. 2019
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import click
+
+from ...testing import get_tox_envs
+from ..console import CONTEXT_SETTINGS, echo_info, echo_warning
+from ..test import test as test_command
+from .start import start
+from .stop import stop
+
+
+@click.command(context_settings=CONTEXT_SETTINGS, short_help='Test an environment')
+@click.argument('checks', nargs=-1)
+@click.option(
+    '--agent',
+    '-a',
+    default='6',
+    help=(
+        'The agent build to use e.g. a Docker image like `datadog/agent:6.5.2`. For '
+        'Docker environments you can use an integer corresponding to fields in the '
+        'config (agent5, agent6, etc.)'
+    ),
+)
+@click.option('--dev/--prod', help='Whether to use the latest version of a check or what is shipped')
+@click.option('--base', is_flag=True, help='Whether to use the latest version of the base check or what is shipped')
+@click.option(
+    '--env-vars',
+    '-e',
+    multiple=True,
+    help=(
+        'ENV Variable that should be passed to the Agent container. '
+        'Ex: -e DD_URL=app.datadoghq.com -e DD_API_KEY=123456'
+    ),
+)
+@click.option('--clean', '-c', is_flag=True, help='Execute setup and tear down actions')
+@click.pass_context
+def test(ctx, checks, agent, dev, base, env_vars, clean):
+    """Test an environment."""
+    check_envs = get_tox_envs(checks, e2e_tests_only=True)
+    tests_ran = False
+
+    # If no checks are specified it means we're testing what has changed compared
+    # to master, probably on CI rather than during local development. In this case,
+    # ensure environments and Agents are spun up and down.
+    if not checks:
+        clean = True
+
+    for check, envs in check_envs:
+        if not envs:
+            echo_warning('No end-to-end environments found for `{}`'.format(check))
+            continue
+
+        # For performance reasons we're generating what to test on the fly and therefore
+        # need a way to tell if anything ran since we don't know anything upfront.
+        tests_ran = True
+
+        for env in envs:
+            if clean:
```

Comment:
I would be tempted to not have that clean option. My suggestion would be to do start/test/stop by default. We can keep `ddev test --e2e` to iterate without setup/teardown of the env, no?

Reply:
The thing is if one of us or a contributor forgets that flag then the env would be town down with a random agent still up, leaving it in an inconsistent state

Reply:
I guess... I'm really against using `clean` though, considering it **starts** the environment.

Reply:
I'm not sure I understand what's the suggested workflow here.
The idea is to do `ddev env test --clean` and that will start, test and stop the env?

I don't understand very well @therve suggested workflow or @ofek's concern here.

Reply:
Note here: As the ideais not to have to run e2e locally for releases anymore I guess the case of start/test/stop is going to become less frequent for developers than it is now

Reply:
I guess that's fine. We can always tweak the process later on. I think detecting if the env is running would be a nice improvement.